### PR TITLE
sd_malloc_total_nbytes() and nbytes api

### PR DESCRIPTION
**Pull request by abheekg**
Incomplete nbytes: I have added sd_malloc_total_nbytes() and tests for it. nbytes tests for hist and rng. The number of nbytes declarations are more. I am working on them. Comments on the other ones awaited.

### CODE REVIEW

File: src/sd_hist.c [hunk 1]
```diff
@@ -208,7 +208,7 @@ static uint32_t nbytes(sd_hist *hist)
 	uint32_t nb;
 	hist_data *h = hist->ptr;
 	nb = sizeof(sd_hist) + sizeof(hist_data);
-	nb += sizeof(uint32_t) * ((h->nu+1) + 3*h->nu + h->nd + h->maxvi);
+	nb += sizeof(uint32_t) * ((h->nu+1) + 3*h->nu + h->nd + (h->maxvi + 1));
```
Comment (maedoc) [inline src/sd_hist.c]: where does the extra +1 come from?

**Comment by maedoc** [comment]
looks good to me! Push commits as you make them and I will continue reviewing.

File: src/sd_hist.c (hunk 1 above)
Comment (abheekg) [inline src/sd_hist.c]: Line #319 `if ((h->vi2i = sd_malloc (sizeof(uint32_t) * (h->maxvi + 1)))==NULL) {`

Comment (maedoc) [inline src/sd_hist.c]: Ah right, because `vi2i`'s last index is `maxvi`, so it's length has to be `maxvi + 1`. Good catch.

**Comment by maedoc** [comment]
I'm closing this because  #143 is adding `n_byte` method for all objects. The `sd_malloc_total_n_byte` would still be a useful addition, so feel free to open a new PR _just_ for that.

**Pull request closed without merging**